Formula/e/envoy.rb

```diff
@@ -47,23 +48,32 @@ class Envoy < Formula
 
   def install
     env_path = "#{HOMEBREW_PREFIX}/bin:/usr/bin:/bin"
+
+    if OS.linux?
+      # Override the default clang to use the one from llvm@18 in PATH in Linux
+      env_path = "#{Formula["llvm@18"].opt_bin}:" + env_path
```

Comment:
I think it might be a good idea to pin to v18 for macOS as well since the official Envoy CI is using that version and we are not building it beyond that in its build pipeline